pkg/devfile/detect.go

```diff
@@ -47,7 +47,14 @@ type AlizerClient struct {
 // Map 3 returns a context to the Dockerfile uri or a matched DockerfileURL from the devfile registry if no Dockerfile is present in the context
 // Map 4 returns a context to the list of ports that were detected by alizer in the source code, at that given context
 func search(log logr.Logger, a Alizer, localpath string, devfileRegistryURL string, source appstudiov1alpha1.GitSource) (map[string][]byte, map[string]string, map[string]string, map[string][]int, error) {
-
+	originalContext := source.Context
+	if originalContext != "" {
+		if originalContext == "./" {
+			originalContext = ""
+		} else if !strings.HasSuffix(originalContext, "/") {
+			originalContext += "/"
+		}
+	}
```

Comment:
do not need this code change.  the context will never be empty. as it's already set in https://github.com/redhat-appstudio/application-service/blob/eeef42302b0bfcac2877973878966f0e880b7fe7/controllers/componentdetectionquery_controller.go#L147-L149

and with the `path.join`, you do not need to append `/` at the end